### DISCUSSION TITLE
UFS: Upload multiple files

### DISCUSSION THREAD

**Discussion by UBWH** [arendst/Tasmota#20370] [Q&A]
My UFS currently looks like this

![image](https://github.com/arendst/Tasmota/assets/72185209/00c0cb09-9e50-4a27-970e-c17a8a60329a)

To backup - I copy one file at a time to my PC
To restore - I copy one file at a time from my PC

Is there a smarter way to Backup/Restore in bulk?

**Comment by gemu2015** [comment]
since you use scripting you may upload multiple files with the external script editor (see docs)

e.g.

UFILES=about.htm,chartEnd.htm,chart_Start.htm, etc

**Comment by UBWH** [comment]
That Script Editor looks perfect for my needs.

But I am confused how I use:
```
UFILES=about.htm,chartEnd.htm,chart_Start.htm
```

- Where do I put that? 
- I see that if I have test.txt that has `>D` and `IP=xx.x.x.x` .. then that file replaces my script.txt on the ESP at xx.x.x.x when I `Window->Run`
- Naturally I don't want to replace my script.txt when uploading other files.

**Reply by gemu2015** [reply]
put UFILES line into your script.
these files are sent together with the script.
there is no option to send UFILES alone.

i edit all my files with the script editor and when i want to upload my helper files i uncomment UFILES
temporarily

**Comment by HofeBY** [comment]
In some PC Backup Software you can add a CMD Script at yor PC.
Into that script you can maybe use "curl" to download some Files from your ESP Tasmota device via http.
Here some examples what to download before Backuproutine PC start´s.
http://IPv4ofYourDevice/ufs/ _matter_device.json
http://192.168.190.21/ufs/YourOwnName.csv
 _persist.json
 script.txt
 .drvset003
 .settings.lkg
 .settings
 .autoconf
 preinit.be
 autoexec.be
 autoexec.bat

**Comment by gemu2015** [comment]
i have added preliminary support for FTP (esp32 and esp8266, (esp8266 very low on RAM))
i will probably make a pr if better testet. i you like to try: https://github.com/gemu2015/Sonoff-Tasmota

#define USE_FTP
#define USER_FTP              "user"
#define PW_FTP                "pw"

must use passive mode
